.cbmc-batch/jobs/aws_cryptosdk_sig_verify_finish/aws_cryptosdk_sig_verify_finish_harness.c

```diff
@@ -23,24 +23,22 @@
 #include <proof_helpers/proof_allocators.h>
 
 void aws_cryptosdk_sig_verify_finish_harness() {
-    /* arguments */
+    /* Nondet input */
     struct aws_cryptosdk_sig_ctx *ctx = can_fail_malloc(sizeof(struct aws_cryptosdk_sig_ctx));
     struct aws_string *signature      = ensure_string_is_allocated_nondet_length();
 
-    /* assumptions */
-    __CPROVER_assume(ctx);
+    /* Assumptions */
+    __CPROVER_assume(ctx != NULL);
```

Comment:
What we really want here is
```c
struct aws_cryptosdk_sig_ctx *ctx = ensure_sig_ctx_has_allocated_members();
```
And inside `ensure_sig_ctx_has_allocated_members` we nondeterministically allocate memory or `ctx`. I'd also expect that `aws_cryptosdk_sig_ctx_is_valid_cbmc` ensures `ctx` is different than NULL, which would avoid the need of the extra assumption. Why do we have a `cbmc` sufix in the name of that function?

Reply:
I've updated the ensure_sig_ctx_has_allocated_members as described and removed the un-needed NULL assumption. I've also realized there is an aws_cryptosdk_sig_ctx_is_valid function which is widely used. I believe the version with the suffix was made earlier. I've made a github issue to replace reconcile the calls to aws_cryptosdk_sig_ctx_is_valid_cbmc.